akka-discovery/src/main/scala/akka/discovery/dns/DnsServiceDiscovery.scala

```diff
@@ -79,8 +79,8 @@ private[akka] class DnsServiceDiscovery(system: ExtendedActorSystem) extends Ser
 
   override def lookup(lookup: Lookup, resolveTimeout: FiniteDuration): Future[Resolved] = {
     lookup match {
-      case Lookup(name, Some(portName), Some(protocol)) ⇒
-        val srvRequest = s"_$portName._$protocol.$name"
+      case l: Lookup if l.portName.isDefined && l.protocol.isDefined ⇒
```

Comment:
Um, maybe this wasn't such a great fit for a pattern match if we're note going to extract options, just if-else it instead, or pat match on the two options?

Reply:
I agree, makes sense as `if` now. updated